### PR TITLE
refactor wranglerjs installation

### PR DESCRIPTION
**Pull request by xtuc**
Downloads a wranglerjs release from GitHub and install it in {Wrangler}'s cache;
globally.

Closes https://github.com/cloudflare/wrangler/issues/133

This change also contains a few {wranglerjs} renames.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ categories = ["wasm"]
 
 [dependencies]
 atty = "0.2.11"
-binary-install = "0.0.2"
+binary-install = { git = "https://github.com/xtuc/binary-install.git", branch = "feat-download-artifact" }
```
Comment (xtuc) [inline Cargo.toml]: https://github.com/rustwasm/binary-install/pull/1

File: src/wranglerjs/mod.rs [hunk 2]
```diff
@@ -177,29 +183,39 @@ pub fn run_npm_install() -> Result<(), failure::Error> {
...
-        failure::bail!("failed to execute `{:?}`: exited with {}", command, status)
-    }
+        let tool_name = "wranglerjs";
+        let version = env!("CARGO_PKG_VERSION");
+        let wranglerjs_path = install::install_artifact(tool_name, "cloudflare", cache, version)?;
+        info!("wranglerjs downloaded at: {:?}", wranglerjs_path.path());
+        wranglerjs_path.path()
+    };
+
+    run_npm_install(wranglerjs_path.clone()).expect("could not install wranglerjs dependecies");
```
Comment (signalnerve) [inline src/wranglerjs/mod.rs]:
```suggestion
    run_npm_install(wranglerjs_path.clone()).expect("could not install wranglerjs dependencies");
```

File: build.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+use std::env;
```
[the remark this answers is not in the document]
Comment (xtuc) [reply build.rs]: two things:
- inject the cargo profile (https://github.com/cloudflare/wrangler/pull/163/files#diff-c48b9d7c36228e6d93dce9d05689c7abR9), in dev mode we don't want to install wranglerjs from the latest release on GitHub, but rather use the version in the source directly.
- we need the location to the source in order to absolutly point to wranglerjs, it's used by the previous point.

**Comment by xtuc** [comment]
Closing in favor of #173 and #174; please review.

**Pull request closed without merging**